arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/RaiseAccumulate.kt

```diff
@@ -521,10 +520,23 @@ public inline fun <Error, A> Raise<Error>.forEachAccumulating(
   iterator: Iterator<A>,
   combine: (Error, Error) -> Error,
   @BuilderInference block: RaiseAccumulate<Error>.(A) -> Unit
+): Unit = forEachAccumulatingImpl(iterator, combine, block, block)
+
+@PublishedApi
+internal inline fun <Error, A> Raise<Error>.forEachAccumulatingImpl(
+  iterator: Iterator<A>,
+  combine: (Error, Error) -> Error,
+  @BuilderInference blockUntilError: RaiseAccumulate<Error>.(A) -> Unit,
+  @BuilderInference blockAfterError: RaiseAccumulate<Error>.(A) -> Unit
 ) {
   var error: Any? = EmptyValue
+  var errorArose = false
```

Comment:
Nit: Can't this be worked out from `error == EmptyValue`? Not sure that either way is going to be faster than the other or anything